tutorials/shaders/compute_shaders.rst

```diff
@@ -0,0 +1,242 @@
+A storage buffer takes an array of bytes and allows the CPU to transfer data to and from the GPU.
+
+So let's initialize an array of doubles and create a storage buffer:
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    # Prepare our data. We use doubles in the shader, so we need 64 bit.
+    var input := PackedFloat64Array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
+    var input_bytes := input.to_byte_array()
+    
+    # Create a storage buffer that can hold our double values.
+    # Each double has 8 byte (64 bit) so 10 x 8 = 80 bytes
+    var buffer := rd.storage_buffer_create(input_bytes.size(), input_bytes)
+    
+ .. code-tab:: csharp
+
+    // Prepare our data. We use doubles in the shader, so we need 64 bit.
+    var input = new double[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
+    var inputBytes = new byte[input.Length * sizeof(double)];
+    Buffer.BlockCopy(input, 0, inputBytes, 0, inputBytes.Length);
+    
+    // Create a storage buffer that can hold our double values.
+    // Each double has 8 byte (64 bit) so 10 x 8 = 80 bytes
+    var buffer = rd.StorageBufferCreate((uint)inputBytes.Length, inputBytes);
+
+With the buffer in place we need to tell the rendering device to use this buffer.
+To do that we will need to create a uniform (like in normal shaders) and assign it to a uniform set which we can pass to our shader later.
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    # Create a uniform to assign the buffer to the rendering device
+    var uniform := RDUniform.new()
+    uniform.uniform_type = RenderingDevice.UNIFORM_TYPE_STORAGE_BUFFER
+    uniform.binding = 0
+    uniform.add_id(buffer)
+    var uniform_set := rd.uniform_set_create([uniform], shader, 0)
+    
+ .. code-tab:: csharp
+ 
+    // Create a uniform to assign the buffer to the rendering device
+    var uniform = new RDUniform
+    {
+        UniformType = RenderingDevice.UniformType.StorageBuffer,
+        Binding = 0
+    };
+    uniform.AddId(buffer);
+    var uniformSet = rd.UniformSetCreate(new Array<RDUniform> { uniform }, shader, 0);
+
+
+Defining a compute pipeline
+---------------------------
+The next step is to create a range of instructions our GPU can execute.
+We need a pipeline and a compute list for that.
+
+The steps we need to do to compute our result are: 
+
+1. Create a new pipeline.
+2. Begin a list of instructions for our GPU to execute.
+3. Bind our compute list to our pipeline
+4. Bind our buffer uniform to our pipeline
+5. Execute the logic of our shader
+6. End the list of instructions
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    # Create a compute pipeline
+    var pipeline := rd.compute_pipeline_create(shader)
+    var compute_list := rd.compute_list_begin()
+    rd.compute_list_bind_compute_pipeline(compute_list, pipeline)
+    rd.compute_list_bind_uniform_set(compute_list, uniform_set, 0)
+    rd.compute_list_dispatch(compute_list, 5, 1, 1)
```

Comment:
If we launch 5 work groups with 4 invocations each we will get 20 compute shader invocations and ``gl_GlobalInvocationID.x`` will range from 0-19 inclusive. Since we only have 10 elements in our storage buffer we will be reading from and writing to memory outside the buffer which may cause this to fail on some hardware. I suggest changing the number of local invocations to 2

Reply:
I adjusted this part a little bit to include the information about accessing memory outside of your control. I think this is a very important missing detail I also wasn't aware of.